assets/src/components/mapPage.tsx

```diff
@@ -0,0 +1,126 @@
+import { Socket } from "phoenix"
+import React, { ReactElement, useContext, useState } from "react"
+import { SocketContext } from "../contexts/socketContext"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import useSearchResults from "../hooks/useSearchResults"
+import { useTripShape } from "../hooks/useShapes"
+import { isVehicle } from "../models/vehicle"
+import { Vehicle, VehicleOrGhost } from "../realtime"
+import { SearchPageState } from "../state/searchPageState"
+import Map from "./map"
+import RecentSearches from "./recentSearches"
+import SearchForm from "./searchForm"
+import SearchResults from "./searchResults"
+
+enum MobileDisplay {
+  List = 1,
+  Map,
+}
+
+const thereIsAnActiveSearch = (
+  vehicles: VehicleOrGhost[] | null,
+  searchPageState: SearchPageState
+): boolean => vehicles !== null && searchPageState.isActive
+
+const filterVehicles = (
+  vehiclesOrGhosts: VehicleOrGhost[] | null
+): Vehicle[] => {
+  return vehiclesOrGhosts === null
+    ? []
+    : (vehiclesOrGhosts.filter((vog) => isVehicle(vog)) as Vehicle[])
```

Comment:
**suggestion(non-blocking):** this is another place where it would be nice to get rid of the `as` if possible.